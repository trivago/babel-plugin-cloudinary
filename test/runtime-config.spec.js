const babel = require("babel-core");
const plugin = require("../src");

jest.mock(
  "../.cloudinaryrc",
  () => ({
    native: {
      // eslint-disable-next-line
      cloud_name: "trivago",
      secure: true,
    },
    host: "trivago.images.com",
    overrideBaseUrl: true,
  }),
  { virtual: true }
);

describe("runtime-config", () => {
  describe("when `overrideBaseUrl` is set to true", () => {
    describe("and `host` is provided", () => {
      it("should override the base URL with the provided host", () => {
        const input = "const imageUrl = __buildCloudinaryUrl('my-picture', {});";
        const { code } = babel.transform(input, { plugins: [plugin] });

        expect(code).toMatchSnapshot();
      });
    });

    // describe("and `host` is not provided", () => {
    //   it("should throw an error", () => {
    //     const input = "const imageUrl = __buildCloudinaryUrl('my-picture', {});";
    //     const { code } = babel.transform(input, { plugins: [plugin] });

    //     expect(code).toMatchSnapshot();
    //   });
    // });
  });
});
