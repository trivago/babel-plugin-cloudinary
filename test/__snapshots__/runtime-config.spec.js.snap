// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime-config when \`overrideBaseUrl\` is set to true and \`host\` is provided should override the base URL with the provided host 1`] = `"const imageUrl = \`\${'https://trivago.images.com/'}\${'my-picture'}\${'.jpeg'}\${}\${}\`;"`;
